components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java

```diff
@@ -127,6 +128,47 @@ void addMonetizationUsagePublishInfo(MonetizationUsagePublishInfo monetizationUs
     void updateMonetizationUsagePublishInfo(MonetizationUsagePublishInfo monetizationUsagePublishInfo)
             throws APIManagementException;
 
+    /**
+     * Add a bot detection alert subscription
+     *
+     * @param email email to be registered for the subscription
+     * @throws APIManagementException if an error occurs when adding a bot detection alert subscription
+     */
+    void addBotDetectionAlertSubscription(String email) throws APIManagementException;
+
+    /**
+     * Delete a bot detection alert subscription
+     *
+     * @param uuid uuid of the subscription
+     * @throws APIManagementException if an error occurs when deleting a bot detection alert subscription
+     */
+    void deleteBotDetectionAlertSubscription(String uuid) throws APIManagementException;
+
+    /**
+     * Retrieve a bot detection alert subscription by querying a particular field (uuid or email)
+     *
+     * @param field field to be queried to obtain the bot detection alert subscription. Can be uuid or email
+     * @param value value corresponding to the field (uuid or email value)
+     * @return if a subscription exists, returns the bot detection alert subscription, else returns a null object
+     * @throws APIManagementException if an error occurs when retrieving a bot detection alert subscription
```

Comment:
Needs to format. (add spaces in betwee param name and the description so that all the descriptions are aligned on same point)

Reply:
The parameter name and description are separated using a single space and formatting this section using the formatter did not make any changes.